internal/server/dump.go

```diff
@@ -80,6 +114,31 @@ func (s *Server) Dump(ctx context.Context, req *daemon.DumpReq) (*daemon.DumpRes
 //// Helper Adapters /////
 //////////////////////////
 
+// Adapter that inserts new adapters after itself based on the type of dump.
+func pluginDumpVMMiddleware(next types.DumpVM) types.DumpVM {
+	return func(ctx context.Context, opts types.Opts, resp *daemon.DumpVMResp, req *daemon.DumpVMReq) (exited chan int, err error) {
+		middleware := types.Middleware[types.DumpVM]{}
+		t := req.GetType()
+		switch t {
+		case "cloud-hypervisor":
```

Comment:
Instead of directly supporting 'cloud-hypervisor' in the daemon, I'd create a plugin for it. Daemon does not need to support any VMs by default, I believe.

And so, the dump handler that is called by the above `DumpVM` will also be provided by the plugin. Each VM's plugin can implement it's own handler.